### PR TITLE
KAFKA-13035  updated documentation for connector restart REST API to …

### PR DESCRIPTION
**Pull request by kpatelatwork**
…include the tasks restart behavior

Targets AK 3.0.0 where KAFKA-4793 will get released.

@kkonstantine and @rhauch Could you please review and see if the changes look good.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by kpatelatwork** [comment]
This is how formatting looks locally
image

**Review by showuon — APPROVED** [review]
LGTM! Thanks for the PR.

File: docs/connect.html [hunk 1]
```diff
@@ -310,7 +310,12 @@ <h4><a id="connect_rest" href="#connect_rest">REST API</a></h4>
         <li><code>GET /connectors/{name}/tasks/{taskid}/status</code> - get current status of the task, including if it is running, failed, paused, etc., which worker it is assigned to, and error information if it has failed</li>
         <li><code>PUT /connectors/{name}/pause</code> - pause the connector and its tasks, which stops message processing until the connector is resumed</li>
         <li><code>PUT /connectors/{name}/resume</code> - resume a paused connector (or do nothing if the connector is not paused)</li>
-        <li><code>POST /connectors/{name}/restart</code> - restart a connector (typically because it has failed)</li>
+        <li><code>POST /connectors/{name}/restart?includeTasks=&lt;true|false&gt;&amp;onlyFailed=&lt;true|false&gt;</code> - restart a connector and its tasks instances.
+            <ul>
+                <li>the "includeTasks" parameter specifies whether to restart the connector instance and task instances ("includeTasks=true") or just the connector instance ("includeTasks=false"), and defaults to "false".</li>
```
Comment (rhauch) [inline docs/connect.html]: Let's mention maintaining backward compatibility:
```suggestion
                <li>the "includeTasks" parameter specifies whether to restart the connector instance and task instances ("includeTasks=true") or just the connector instance ("includeTasks=false"), with the default ("false") preserving the same behavior as earlier versions.</li>
```

File: docs/connect.html [hunk 2]
```diff
@@ -310,7 +310,12 @@ <h4><a id="connect_rest" href="#connect_rest">REST API</a></h4>
...
+                <li>the "onlyFailed" parameter specifies whether to restart just the instances with a FAILED status ("onlyFailed=true") or all instances ("onlyFailed=false"), and defaults to "false".</li>
```
Comment (rhauch) [inline docs/connect.html]:
```suggestion
                <li>the "onlyFailed" parameter specifies whether to restart just the instances with a FAILED status ("onlyFailed=true") or all instances ("onlyFailed=false"), with the default ("false") preserving the same behavior as earlier versions.</li>
```

**Review by rhauch — COMMENTED** [review]
Thanks, @kpatelatwork. A minor suggestion to mention how this feature by default preserves the same behavior as earlier versions.

File: docs/connect.html (hunk 2 above)
Comment (kpatelatwork) [reply docs/connect.html]: Fixed

File: docs/connect.html (hunk 1 above)
Comment (kpatelatwork) [reply docs/connect.html]: Fixed

**Comment by kpatelatwork** [comment]
@rhauch great call about the backward compatibility note, I applied the suggestions. When you get time, could you please check again if it looks good?

**Review by rhauch — APPROVED** [review]
LGTM. Thanks, @kpatelatwork!

**Pull request merged**